tfjs-backend-webgpu/src/kernels/BroadcastArgs.ts

```diff
@@ -0,0 +1,55 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =============================================================================
+ */
+
+import {backend_util, BroadcastArgs, BroadcastArgsInputs, KernelConfig, KernelFunc, TensorInfo, TypedArray, util} from '@tensorflow/tfjs-core';
+
+import {WebGPUBackend} from '../backend_webgpu';
+import {BroadcastArgsProgram} from '../broadcast_args_webgpu';
+
+export function broadcastArgs(args: {
+  inputs: BroadcastArgsInputs,
+  backend: WebGPUBackend,
+}): TensorInfo {
+  const {inputs, backend} = args;
+  const {s0, s1} = inputs;
+
+  if (backend.shouldExecuteOnCPU([s0, s1])) {
+    const s0TensorInfo = backend.tensorMap.get(s0.dataId);
```

Comment:
Why don't we reuse the CPU impl?

Reply:
1. cpu backend does not export the CPU impl for other backends.
2. cpu backend also calls the tfjs-core common function to implement the feature, it is better for other backends to call the tfjs-core common function directly.

Reply:
Per discussion, a similar interface (xxxImplCPU) to be used as CPU fallback is a good approach. However, this may need some changes at CPU backend, so let's refine this in a future PR. LGTM with this one.